opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
Oh, note that actually this optimization would not be facilitated with more const support (as I've just realized). I'm thinking perhaps the only think that would go wrong is something with multithread support, but if that were the case I don't see why.